src/gobstuf/stuf/brp/base_request.py

```diff
@@ -55,8 +55,6 @@ def _set_values(self, values: dict):
         :param values:
         :return:
         """
-        assert values.keys() == self.parameter_paths.keys()  # Should never fail
```

Comment:
Why is this line removed?

Reply:
Ah, I see now why it's removed. Shouldn't we just change this check instead of removing it?

Reply:
Because every argument combination is already checked on receipt.
So the argument combination is already checked for validity
And not all parameters are required. Combination bsn requires 1 parameter, postcode, buisnummer 2 parameters. But not all have to be specified.
The assertion will fail because there are 3 possible parameters (bsn, postcode, huisnummer) but not all are available.
But please suggest another option if I have missed something.

Reply:
Well, it has already been checked when the combination was checked so why check it twice?

Reply:
Development errors. This class assumes (correctly, at this moment) that the calling class has checked the parameter combinations. That is some external knowledge that's verified by the assertion. You and I know at this moment how that interaction works, but it's easy to make a mistake there in the future. This assertion would fail at the first run that this precondition is broken. Yes, eventually the code would probably fail further down the line, but that is unpredictable. I think an assertion makes it clear as early as possible that a developer did something wrong (instead of generating a cryptic error message further down the line that eventually resolves to the same problem)

Reply:
OK, point taken, I will update the assertion...